lib/pf/api.pm

```diff
@@ -222,9 +223,29 @@ sub firewallsso : Public {
 
     my $logger = pf::log::get_logger();
 
-    foreach my $firewall_conf ( sort keys %pf::config::ConfigFirewallSSO ) {
-        my $firewall = pf::factory::firewallsso->new($firewall_conf);
-        $firewall->action($firewall_conf,$postdata{'method'},$postdata{'mac'},$postdata{'ip'},$postdata{'timeout'});
+    foreach my $firewall_id ( sort keys %pf::config::ConfigFirewallSSO ) {
+        my $firewall = pf::factory::firewallsso->new($firewall_id);
+
+        if($postdata{method} eq "Update"){
+            if( pf::util::isenabled($pf::config::ConfigFirewallSSO{$firewall_id}{cache_updates}) ){
+                my $cache = pf::CHI->new( namespace => 'firewall_sso' );
+                my $cache_timeout = $pf::config::ConfigFirewallSSO{$firewall_id}{cache_timeout} || $postdata{timeout} / 2;
```

Comment:
Why not mutiply by 2 ? and instead of just caching $firewall_id -$postdata{ip} key why don't you cache $firewall_id -$postdata{ip} - $postdata{mac}

So if we receive a new dhcp request after the dhcp lease timeout  then the device will be still in the cache.